### ISSUE TITLE
Compile for older Java Runtimes

### ISSUE THREAD

**Issue by axkr** [iczelia/Maja#3]
Can you please compile your library for older runtimes.

I get this error in an attempt to use the library under Java 11:
```
java.lang.UnsupportedClassVersionError: rocks/palaiologos/maja/Maja has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```

**Comment by iczelia** [comment]
PRs welcome, and I will be merge and publish them on Maven :).

I am currently unable to help too much with open source myself right now because of my exam period and other things piling up.

**Comment by axkr** [comment]
I got these "incompatibilities" from the java 17 "new features" `record`and `pattern matching in instanceof` statement.

Do you want to stick with Java17 or should I rewrite to  older syntax?
*Note:* I need Java 11 because of Android system (API 33)

```
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/Pair.java:[11,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/Hypergeometric.java:[628,28] switch rules are not supported in -source 11
  (use -source 14 or higher to enable switch rules)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/matrix/DoubleEigenvalueDecompositionResult.java:[23,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/expression/DefaultExpressionVisitor.java:[79,42] pattern matching in instanceof is not supported in -source 11
  (use -source 16 or higher to enable pattern matching in instanceof)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/matrix/DoubleLUDecompositionResult.java:[12,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/Complex.java:[9,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/matrix/DoubleSVDResult.java:[15,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/matrix/DoubleCholeskyDecompositonResult.java:[10,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
[ERROR] /C:/Users/dev/git/Maja/src/main/java/rocks/palaiologos/maja/Dual.java:[9,8] records are not supported in -source 11
  (use -source 16 or higher to enable records)
```

**Comment by iczelia** [comment]
Ideally the repo would use Java17 syntax. If you have a way to make it compatible with J11 or J8, I will merge it.

If there is no other way then I suppose that the usages of Java17 could be replaced with something backwards-compatible.

**Comment by axkr** [comment]
I think it's too early to use Java 17 in our project. 
Maybe we should keep this issue "on hold" for a later consideration.

**Comment by iczelia** [comment]
Fixed in v0.2.3.

**Comment by axkr** [comment]
> Fixed in v0.2.3.

Thanks for the change. Can we talk on Discord?
- https://discord.gg/tYknzr2qam
